### DISCUSSION TITLE
Devices shows up as 4 controllers in host mode.

### DISCUSSION THREAD

**Discussion by multigamesystem** [OrangeFox86/DreamPicoPort#109] [Q&A]
How do I stop the PICO from being recognized as 4 controllers?  I only have one controller plugged in and this is causing issues with RetroArch.

**Comment by Tails86** [comment]
At the moment, there is nothing coded to easily accommodate for this. Editing `src/hal/Usb/Client/Hid/usb_descriptors.h` and `src/hal/Usb/Client/Hid/usb_descriptors.c` for a single controller would be the best solution at the moment. The value of NUMBER_OF_GAMEPADS should then flow down to the necessary components in main. 

I have been caught up with trying to figure out what the best way to move forward with this project. It may simply be to support only 1 controller anyway so this can conform to XInput rather than try to use the HID spec which I have run into trouble with.

**Reply by multigamesystem** [reply]
Do you have a version of the .uf2 with just one controller implemented?

**Reply by Tails86** [reply]
I made a branch for that here https://github.com/OrangeFox86/DreamcastControllerUsbPico/tree/single-controller
turns out that there was one other file to touch in order to do this.

I can't upload a .uf2 file here, so just remove the .txt part of the extension in the following.
[host-1-player.uf2.txt](https://github.com/OrangeFox86/DreamcastControllerUsbPico/files/13232489/host-1-player.uf2.txt)

**Reply by multigamesystem** [reply]
Thanks much ^__^

**Reply by multigamesystem** [reply]
*Fixed

![RASP_PI_PICO_pinout-OrangeFox86](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/8bc0b266-c35c-4ef0-b57b-03832d0e5b73)

**Reply by Tails86** [reply]
Looking good! I like it. I noticed a couple of things with the pin labels:

- Pin 5 is SDCKB (not A)
- Some other projects use pin 4 as Sense which is why it seems this pin is often labeled as a sense pin online, but it's really just another ground wire for the controller. As far as I can tell, the Dreamcast itself ties pins 3 and 4 to ground (ohm meter across those 2 pins reads near 0-ohms). The Dreamcast instead continually pings the controller to detect presence, which is what this project mimics.

**Reply by multigamesystem** [reply]
Also this adapter was made for the Dreamcast Mini Ive been designing and building. 
![1698899474334](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/35f4490e-89ff-4f78-8a8a-3fa61e6aa03f)
![1699246076729](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/aa247b61-348a-4dd5-95cf-e7509e9025d7)
![1699246076959](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/12a08a36-a19f-402d-a845-ebc529c1d26d)
![1699246076977](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/430377b6-c06d-479b-b027-745eadedab93)
![1699246077040](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/9af8f74d-09fc-4c77-815b-e5995aa76fec)

**Reply by Tails86** [reply]
Holy crap, dude. That's amazing! Is this a DIY thing or are you making a sellable product?

**Reply by ErykDarnowski** [reply]
That's so cool! Did you use USB type Bs to make a small imitation of the controller ports?

**Reply by multigamesystem** [reply]
Yeah it was the closest to a Dreamcast plug I could get at this scale.  Although board mount USB B jacks are actually upside down compared to the Dreamcast plugs so it didnt work out as great as I had hoped for my PCB version.  
![IMG_20231030_152815](https://github.com/OrangeFox86/DreamcastControllerUsbPico/assets/49738515/c7c15a6d-be6a-4b74-9f5d-462f9d20cb35)

**Reply by multigamesystem** [reply]
I designed and prototyped these from scratch myself.  I do sell them at a limited capacity since Im one person 3D printing out of my basement workshop.
